File: Materiales/MinticG72Ciclo1/Repaso/Ejercicio3.py
"""
Carlos A Delgado S
23 de Junio de 2022

Una empresa requiere un programa para gestionar sus facturas, estas
facturas manejan un código de producto y la cantidad comprada.

Los productos que vende la empresa son:

Codigo  Descripción  Valor unitario
100     Camisa           10000
200     Pantalón         30000
300     Medias           10000
400     Saco              8000

Estos precios no incluyen el IVA 19%. La información está codificada en una lista de tuplas, donde cada tupla tiene el código y la cantidad vendida.
[(100,10), (200,20),(300,10),(400,20),(100,20)]
La función debe retornar un diccionario que tenga la siguiente estructura
{"subtotal":subtotal,"iva":valor_iva,"total":total}
El valor del iva y el total redondearlo con dos decimales round(...,2)
Para el ejemplo anterior se retorna {"subtotal":1160000,"iva":220400,"total":1380400}

def generar_venta(lst_ventas):
    pass #Esto se debe de borrar

"""
def obtener_valor(cod):
    if cod==100:
        return 10000
    elif cod==200:
        return 30000
    elif cod==300:
        return 10000
    elif cod==400:
        return 8000
    else: #Si alguien metio un codigo mal
        return 0

def generar_venta(lst_ventas):
    salida = {} #Inicializo un diccionario vacio
    subtotal = 0
    valor_iva = 0
    total = 0
    for venta in lst_ventas:
        subtotal += obtener_valor(venta[0])*venta[1]

    salida["subtotal"] = subtotal
    valor_iva= round(subtotal*0.19,2)
    salida["valor_iva"] = valor_iva
    total = round(subtotal+valor_iva,2)
    salida["total"] = total
    return salida

print(generar_venta([(100,10), (200,20),(300,10),(400,20),(100,20)]))

File: Materiales/MinticG72Ciclo1/Repaso/Ejercicio3.java
/*
 * Carlos A Delgado
 * 23 de junio de 2022
 * Ejercicio3 hecho en Java
 */

import java.util.*;

public class Ejercicio3 {
    public static void main(String[] args) {
        Solucion objSolucion = new Solucion();
        int[][] lista = {{100,10}, {200,20},{300,10},{400,20},{100,20}};
        ArrayList<ArrayList<Integer> > lista_venta =  new ArrayList<ArrayList<Integer>>();

        for(int[] element: lista){
            ArrayList<Integer> aux = new ArrayList<Integer>();
            aux.add(element[0]);
            aux.add(element[1]);
            lista_venta.add(aux);

        }
        System.out.println(objSolucion.generar_venta(lista_venta));
    }
}

class Solucion{

    public int obtener_valor(int cod){
        if(cod==100)
            return 10000;
        else if(cod==200)
            return 30000;
        else if(cod==300)
            return 10000;
        else if(cod==400)
            return 8000;
        else
            return 0;
    }

    public HashMap<String,Double> generar_venta(ArrayList<ArrayList <Integer>> lst_ventas){
        HashMap<String,Double> salida = new  HashMap<String,Double>();
        double subtotal = 0;
        double valor_iva = 0;
        double total = 0;
        for(ArrayList <Integer> venta : lst_ventas){
            subtotal += obtener_valor(venta.get(0))*venta.get(1);
        }
            
        salida.put("subtotal",subtotal);
        valor_iva= Math.round(subtotal*0.19);
        salida.put("iva",valor_iva);
        total = Math.round(subtotal+valor_iva);
        salida.put("total",total);
        return salida;
    }
}

//{'subtotal': 1160000, 'valor_iva': 220400.0, 'total': 1380400.0}

